addons/udes_stock/tests/test_stock_warehouse.py

```diff
@@ -7,22 +7,37 @@ def setUpClass(cls):
         super(TestStockWarehouseModel, cls).setUpClass()
         cls.Warehouse = cls.env["stock.warehouse"]
 
-    def test01_get_test_picking_types(self):
+    def test_get_test_picking_types(self):
         """Test checks if the test picking types are present"""
         test_picking_types = self.warehouse.get_picking_types().filtered(
             lambda pt: "TEST" in pt.name
         )
         self.assertEqual(len(test_picking_types), 6)
 
-    def test02_get_picking_types(self):
+    def test_get_picking_types(self):
         """Test checks if the test picking types and the normal pick types are present"""
         test_picking_types = self.warehouse.get_picking_types()
         self.assertEqual(len(test_picking_types), 9)
 
-    def test03_get_default_picking_types_new_warehouse(self):
+    def test_get_default_picking_types_new_warehouse(self):
         """Create a new warehouse and look at the default picking types"""
         # Create a new company creates a new warehouse
         test_company = self.create_company("test_company")
         new_warehouse = self.Warehouse.search([("company_id", "=", test_company.id)])
         test_picking_types = new_warehouse.get_picking_types()
         self.assertEqual(len(test_picking_types), 3)
+    
+    def test_check_picking_types_prefix_after_updating_warehouse_name(self):
+        """
+        Test that making changes to the warehouse does not cause the pickingtype prefixes to reset. 
+        """ 
+        Pickingtype = self.env["stock.picking.type"]
+        
+        goods_in_prefix = Pickingtype.search([("name", "=", "Goods In")]).sequence_id.prefix
+        self.assertEqual(goods_in_prefix, "IN")
+        
+        self.warehouse.write({"code":"NEW_WAREHOUSE_CODE"})
+
+        goods_in_prefix = Pickingtype.search([("name", "=", "Goods In")]).sequence_id.prefix
+        self.assertEqual(goods_in_prefix, "IN")
+
```

Comment:
Might be nice to have a test with  a new warehouse in as well

Reply:
Upon creating a new warehouse, Odoo will automatically create picking types with the Odoo style prefix WH/OUT and sequences WH/OUT/XXXX1. We set our own sequences and prefixes in data, in UDES14. For example we set picking type goods in to have prefix IN which results in a sequence INXXXX1 for example.

As it will never be a warehouse user creating a picking type and always a support user or dev, it should not matter if these picking types are created with the wrong sequence and prefix as we will always set our own.

Hence the test of creating a new warehouse would mean that I would have to reset all the sequences and do a similar test to what I have already done, which feels a bit redundant to me.